roles/polarion/tasks/main.yml

```diff
@@ -108,10 +123,14 @@
           source "{{ cifmw_polarion_jump_repo_dir }}/jump-venv/bin/activate" &&
           {{ cifmw_polarion_jump_repo_dir }}/jump-venv/bin/python jump.py
           --testrun-id={{ cifmw_polarion_testrun_id }}
-          --xml-file={{ cifmw_polarion_jump_result_dir }}/results_merged.xml
+          --testrun-title={{ cifmw_polarion_testrun_title }}-{{ loop_idx }}
+          --xml-file={{ item.path }}
           --update_testcases={{ cifmw_polarion_update_testcases | default(false) }}
-          --custom-fields cijoburl={{ cifmw_zuul_url }}/{{ zuul.build }}
+          --custom-fields cijoburl={{ cifmw_zuul_url }}/{{ zuul.build }},jobteststage={{ result_dirs.files[ loop_idx ].path | basename }}
           {{ cifmw_polarion_jump_extra_vars | default ('') }}
+      loop: "{{ merged_xml_files.files }}"
+      loop_control:
+        index_var: loop_idx
```

Comment:
@kopecmartin can you please have a look at my comment there https://issues.redhat.com/browse/OSPCIX-470?focusedId=25560054&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-25560054

I think this patch introduced a new bug.

if needed I can file a dedicated CIX for that (the referenced cix is tracking a different polarion reporting issue)

Reply:
thanks for bringing this up, fix has been proposed - https://github.com/openstack-k8s-operators/ci-framework/pull/2345